tests/testthat/test-derive_var_confirmation_flag.R

```diff
@@ -0,0 +1,183 @@
+    "3",      1,        "SD",   NA_character_,
+    "4",      1,        "PR",   "Y",
+    "4",      2,        "PD",   NA_character_,
+    "4",      3,        "SD",   NA_character_,
+    "4",      4,        "SD",   NA_character_,
+    "4",      5,        "PR",   NA_character_
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## Test 2: filter with first_cond ----
+test_that("derive_var_confirmation_flag Test 2: filter with first_cond", {
+  actual <-
+    derive_var_confirmation_flag(
+      data,
+      new_var = CONFFL,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVALC),
+      first_cond = AVALC == "CR" &
+        AVALC.join == "CR",
+      order = vars(AVISITN),
+      filter = TRUE
+    )
+
+  expected <- tribble(
+    ~USUBJID, ~AVISITN, ~AVALC, ~CONFFL,
+    "1",      1,        "PR",   NA_character_,
+    "1",      2,        "CR",   "Y",
+    "1",      3,        "CR",   NA_character_,
+    "1",      4,        "SD",   NA_character_,
+    "1",      5,        "NE",   NA_character_,
+    "2",      1,        "SD",   NA_character_,
+    "2",      2,        "PR",   NA_character_,
+    "2",      3,        "PD",   NA_character_,
+    "3",      1,        "SD",   NA_character_,
+    "4",      1,        "PR",   NA_character_,
+    "4",      2,        "PD",   NA_character_,
+    "4",      3,        "SD",   NA_character_,
+    "4",      4,        "SD",   NA_character_,
+    "4",      5,        "PR",   NA_character_
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## Test 3: filter with first_cond and summary function ----
+test_that("derive_var_confirmation_flag Test 3: filter with first_cond and summary function", {
+  actual <-
+    derive_var_confirmation_flag(
+      data,
+      new_var = CONFFL,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVALC),
+      first_cond = AVALC == "PR" &
+        AVALC.join %in% c("CR", "PR"),
+      order = vars(AVISITN),
+      filter = count_vals(AVALC.join, "SD") <= 1,
+      false_value = "N"
+    )
+
+  expected <- tribble(
+    ~USUBJID, ~AVISITN, ~AVALC, ~CONFFL,
+    "1",      1,        "PR",   "Y",
+    "1",      2,        "CR",   "N",
+    "1",      3,        "CR",   "N",
+    "1",      4,        "SD",   "N",
+    "1",      5,        "NE",   "N",
+    "2",      1,        "SD",   "N",
+    "2",      2,        "PR",   "N",
+    "2",      3,        "PD",   "N",
+    "3",      1,        "SD",   "N",
+    "4",      1,        "PR",   "N",
+    "4",      2,        "PD",   "N",
+    "4",      3,        "SD",   "N",
+    "4",      4,        "SD",   "N",
+    "4",      5,        "PR",   "N"
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## Test 4: join_type = "all" ----
+test_that("derive_var_confirmation_flag, Test 4: join_type = 'all'", {
```

Comment:
would be good to see a unit test for each of the other `join_type` options too

Reply:
I think test 5 and 6 are not well designed. I have updated them. Is it clearer now?

Reply:
i think test 6 is incorrect. i don't agree records 1 or 4 should have confirmation flagged, as it seems like the same single record is confirming itself. i always thought you need a separate observation to confirm the original observation.

Reply:
i disagree with this - we're trying to do too many things here. the explanation of the function reads "Derive a flag which depends on **subsequent observations** of the dataset." 

i think we should remove the `at_*` options and the same record shouldn't be able to confirm itself. i don't see that as a realistic use case of this function and it just over-complicates an already complicated function for the users.

Reply:
I agree that the `at_*` options will be used only in rare cases. Nevertheless I would keep them because removing them does not simplify the code, documentation, or usage and in my experience it is just a matter of time until they will be requested.

Reply:
I can see benefit of having `at_*` options, but happy to discuss further at next week meeting

Reply:
Not sure if I understand it 100% correct, so please feel free to correct me if my assessment is based on a wrong assumption.

at_* seems to function as a simple flag based on an occurrence of a single AE, while the purpose of the function is to confirm an AE due to having another AE present.: 

 _Derive a flag which depends on other observations of the dataset. For
#' example, flagging events which need to be confirmed by a second event._

I would keep the functionality down to the purpose of "confirming by a second event", to keep it in line with our manifesto in terms of simplicity and not overload functions with too many options. The function itself seems already be complex in terms of parameters, by adding at_* I feel we actually changing the original meaning of the function itself. 
Here I would really like to keep the function clean to have one purpose, which is confirmation by a second event.

Personally (if I understood the at_* purpose correctly), if I only would like to flag a AE depending on itself, let's say Special AE or a specific AE + Severity or Hospitalization, I would just do this with one line of conditional code: it is simple to read and transparent. 
For example, If I would take over some code and see this function called with the at_* parameter, I first to understand the function and its option and I would always ask myself why somebody used the function, with a lot of additional code, while it only flags this one AE? So it might lead to additional checks which would not be necessary.

Another point might be that if it is rarely used, we should try to avoid these special cases to stay true to our goal of covering 80% and not every possible case. If a functionality like this is really needed, the users could create their own function and submit it. 

These are my arguments for removing the at_* option, if I understood it correctly :)

Reply:
I completely agree that the `at_*` option is needed only in rare case and that the function is quite complex.

I added these options
- because they made the `join_type` parameter complete and
- they do not increase the complexity of the function.

I think we do not gain anything if we remove them. But we risk that users need to implement their own function. We can implement these options for free but if users need to create their own function, it will take them a lot of time.

Reply:
I'd like these `at_` options to be removed please Stefan. It may not increase the complexity from the dev perspective but it massively does from a user perspective as it adds a whole new dimension to what is trying to be achieved here by enabling the original record to confirm itself. If ever the user needs this i prefer that they use a simpler solution themselves as Robin suggests. I already feel like this function will rarely be adopted by users given the concept and documentation is so difficult to get your head around. I think the feedback we're most likely to get is to further simplify it rather than extending it.